src/main/java/com/fulcrumgenomics/util/PickIlluminaIndicesCommand.java

```diff
@@ -184,6 +185,19 @@ Index reverseComplement() {
         }
 
         /**
+         * Checks to make sure tha this object is contained in this set prior to removing.  The object may be removed
+         * by another thread between the containment check and removal, so this method is not guaranteed to be atomic.
+         * If the object is not contained, an exception is thrown.  The value of the remove method is returned.
+         */
+        public boolean containsAndRemove(final Object o) {
```

Comment:
This is unnecessarily complicated.  The `remove()` method already returns a boolean which is true if the value was present and removed, and false if the value was not present.  No need to scan the set twice (and without atomicity).